src/Wrapper/GenericAKVWrapper.php

```diff
@@ -188,12 +226,39 @@ public function decrypt(string $encryptedData): string
         } catch (Throwable $e) {
             throw new ApplicationException('Deciphering failed.', $e->getCode(), $e);
         }
-        $this->verifyMetadata($decryptedContext[self::METADATA_INDEX], $this->metadata);
+        $this->verifyMetadata($decryptedContext[self::METADATA_INDEX], $metadata);
         try {
             $key = Key::loadFromAsciiSafeString($decryptedContext[self::KEY_INDEX]);
             return Crypto::decrypt($encrypted[self::PAYLOAD_INDEX], $key, true);
         } catch (Throwable $e) {
+            $doBackfill = false;
             throw new ApplicationException('Deciphering failed.', $e->getCode(), $e);
+        } finally {
+            if ($doBackfill) {
+                if (isset($this->metadata['stackId']) && self::getTransStackId()) {
+                    $decryptedContext[self::METADATA_INDEX]['stackId'] = self::getTransStackId();
+                }
+                try {
+                    $this->getRetryProxy()->call(function () use ($encrypted, $decryptedContext) {
+                        $context = $this->encode($decryptedContext);
+                        $this->getTransClient()?->setSecret(
+                            new SetSecretRequest($context, new SecretAttributes()),
+                            $encrypted[self::SECRET_NAME],
+                        );
+                    });
+                    $this->logger?->info('Secret "{secretName}" migrated in {stackId} AKV.', [
+                        'secretName' => $encrypted[self::SECRET_NAME],
+                        'stackId' => self::getTransStackId() ?? 'trans',
```

Comment:
to ?? trans je tu k prdu ne, v téhle chvíli musí bejt transStackId nastavený, ne?

Reply:
jinak teda tady až se to bude nasazovat, tak je důležitý zkontrolovat, aby to transstackid bylo správně nastavený, aby se ty šifry založily správně

Reply:
Viz https://github.com/keboola/object-encryptor/pull/52#discussion_r1944522104